extension/githubappauthextension/README.md

```diff
@@ -0,0 +1,51 @@
+# GitHub App Authentication
+
+<!-- markdownlint-disable -->
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [alpha]  |
+| Distributions | [] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Aextension%2Fgithubappauth%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aextension%2Fgithubappauth) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Aextension%2Fgithubappauth%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aextension%2Fgithubappauth) |
+
+[alpha]: https://github.com/open-telemetry/opentelemetry-collector#alpha
+<!-- end autogenerated section -->
+<!-- markdownlint-enable -->
+
+This authentication extension supports authenticating to GitHub by
+authenticating to GitHub as an installation. This was purpose built for the
+GitProviderReceiver to reduce the need for PATs and reduce rate limiting.
+
+There are a few pre-reqs you have to do in order to get a GitHub App to work
+with this extension. These instructions may change based on changes to GitHub.
+
+## Creating a GitHub Application
+
+1. Create a GitHub Application similar to shown in the below pictures.
```

Comment:
Need to update these docs and create a `docs/` dir containing the images I want to leverage.